core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -110,7 +110,7 @@ object RequestChannel extends Logging {
 
     def sizeOfBodyInBytes: Int = bodyAndSize.size
 
-    def sizeInBytes: Int = header.size(new ObjectSerializationCache) + sizeOfBodyInBytes
+    def sizeInBytes: Int = header.size + sizeOfBodyInBytes
```

Comment:
Note for reviewers: 
This is the main change in this PR which impacts performance where we are using the cached value instead of calculating it again.